src/varlinkctl/varlinkctl.c

```diff
@@ -785,7 +785,9 @@ static int verb_call(int argc, char *argv[], void *userdata) {
                                 r = 0;
                         } else {
                                 r = sd_varlink_error_to_errno(error, reply);
-                                if (r != -EBADR)
+                                if (r == -EBADE)
```

Comment:
Instead of conditionalize with `r`, please explicitly check the error message:
```c
if (strv_contains(arg_graceful, error)) {
        ...;
} else if (streq(error, SD_VARLINK_ERROR_EXPECTED_MORE))
        r = log_error_errno(SYNTHETIC_ERRNO(EBADE), "...");
else {
        r = sd_varlink_error_to_errno(error, reply);
        ...;
}
```